src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,222 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private ListNode head;
+    private ListNode tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        ListNode newnode = new ListNode<>(value);
+        if (size == 0) {
+            head = newnode;
+            tail = newnode;
+        } else if (size == 1) {
+            head.setNext(newnode);
+            newnode.setPrev(head);
+            tail = newnode;
+        } else {
+            tail.setNext(newnode);
+            newnode.setPrev(tail);
+            tail = newnode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        ListNode newnode = new ListNode<>(value);
+        if (size == 0 && index == 0) {
+            head = newnode;
+            tail = newnode;
+            size++;
+            return;
+        }
+        if (index == 0 && size > 0) {
+            newnode.setNext(head);
+            head.setPrev(newnode);
+            head = newnode;
+            size++;
+            return;
+        }
+        if (size == index) {
+            tail.setNext(newnode);
+            newnode.setPrev(tail);
+            tail = newnode;
+            size++;
+            return;
+        }
+
+        ListNode temp = findElementByIndex(index);
+        newnode.setNext(temp);
+        newnode.setPrev(temp.getPrev());
+        temp.setPrev(newnode);
+        newnode.getPrev().setNext(newnode);
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        ListNode temp = findElementByIndex(index);
+        return (T) temp.getValue();
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        ListNode temp = findElementByIndex(index);
+        T oldvalue = (T) temp.getValue();
+        temp.setValue(value);
+        return oldvalue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        ListNode temp = findElementByIndex(index);
+        unlink(temp);
+        return (T) temp.getValue();
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        //throw exception if empty ?
```

Comment:
```suggestion
        //throw exception if empty ?
```
i dont know, but when mentors write comments to me, they ask me to remove comments